EDU-3400: Update Nexus docs (#3196)
temporalio/documentation: docs/encyclopedia/nexus-use-cases.mdx

- **Cross-team, cross-domain, and cross-namespace** \-
connect Temporal Applications within and across Namespaces.
- **Share a subset of a Temporal Application** \-
abstract and share a subset of an Application as a Nexus Service.
- **Modular design for growth** \-
modular application design that can evolve as you grow.
- **Smaller failure domains** \- 
each team to have their own Namespace for improved security, troubleshooting, and fault isolation.
- **Multi-region** \-
Nexus requests in Temporal Cloud are routed across a global mTLS-secured Envoy mesh.

:::tip RELATED

- [Evaluate](/evaluate/nexus) why you should use Nexus and learn more about [Nexus use cases](/evaluate/nexus#use-cases).

:::

## Share Workflows Across Namespaces

Nexus is purpose-built to connect Temporal Applications within and across Namespaces.
It addresses the limitations of Child Workflows, Activity Wrappers, and bespoke APIs that target a remote Namespace.
Nexus has a streamlined Temporal developer experience, reliable execution, and integrated observability.

In the past, when a caller Workflow invokes another Workflow directly the caller must know:
- target Workflow's Namesapce and Task Queue.

> - target Workflow Policy options including [Workflow-Id-Reuse-Policy](/workflows#workflow-id-reuse-policy) and [Workflow-Id-Conflict-Policy](/workflows#workflow-id-conflict-policy).

Reviewer: To me it's more about retry policy and timeouts, but yes also the ID reuse policy.

Author: Added those.

This creates a high degree of coupling between the caller and handler, by exposing internal implementation details to the caller.
This adds friction for the caller, who shouldn't need to know this level of detail.
It's more difficult to refactor or migrate handler Workflows to different Namespace or Task Queue.
In short, Workflow to Workflow is a leaky abstraction.

Nexus addresses this by providing a cleaner service contract between the caller and handler.
Nexus is suitable for abstracting and sharing Workflows across team, domain, and namespace boundaries.
Nexus requests in Temporal Cloud are routed across a global mTLS-secured Envoy mesh, so they're also suitable for multi-region use cases.
 
Enable calls across Namespaces by:

1. Creating a [Nexus Endpoint](/nexus/endpoints) in the [Nexus Registry](/nexus/registry) that:
   1. Targets the handler Namespace.
   2. Allows the caller Namespace.
2. Creating a [Nexus Service](/nexus/services) in a Worker within a handler Namespace.
   1.  Abstract Workflows with Nexus Operations [using Temporal SDK builder functions for Nexus Operations](/nexus/operations#sdk-support).

>    2.  Register the Nexus Service with the Worker

Reviewer: ```suggestion
   2.  Register the Nexus Service with the Worker.
```

:::

[Temporal Nexus](/evaluate/nexus) enables the following use cases:

- **Cross-team, cross-domain, and cross-namespace** \-
connect Temporal Applications within and across Namespaces.
- **Share a subset of a Temporal Application** \-
abstract and share a subset of an Application as a Nexus Service.
- **Modular design for growth** \-
modular application design that can evolve as you grow.
- **Smaller failure domains** \- 
each team to have their own Namespace for improved security, troubleshooting, and fault isolation.
- **Multi-region** \-
Nexus requests in Temporal Cloud are routed across a global mTLS-secured Envoy mesh.

:::tip RELATED

- [Evaluate](/evaluate/nexus) why you should use Nexus and learn more about [Nexus use cases](/evaluate/nexus#use-cases).

:::

## Share Workflows Across Namespaces

Nexus is purpose-built to connect Temporal Applications within and across Namespaces.
It addresses the limitations of Child Workflows, Activity Wrappers, and bespoke APIs that target a remote Namespace.
Nexus has a streamlined Temporal developer experience, reliable execution, and integrated observability.

> In the past, when a caller Workflow invokes another Workflow directly the caller must know:

Reviewer 2: ```suggestion
Without Nexus, when a caller Workflow invokes another Workflow directly the caller must know:
```
